### PR TITLE
Add form archiving and unarchiving

### LINKED ISSUE #631: Unjournal: remove outdated forms
## Motivation
Clean up the Unjournal community's unnecessary forms from testing.

## Requirements
- Re-introduce the Archive button to the community forms page.
  - There is a comment that flagposts where to do this.
- Archiving a form should move it into the "Archived" tab.
- Add an Un-archive button to the archived tab that lets a user un-archive a form.
- Slugs of archived forms are still reserved (until we add a proper delete button)
- Fill page for archived forms continue to work
- Form link directives in templates for archived forms continue to work

## Who Requested
David / Unjournal

### PR DESCRIPTION
**Pull request by kalilsn**
## Issue(s) Resolved
Resolves #631 
## High-level Explanation of PR
- Adds archive button back
- Adds restore button

### CODE REVIEW

File: core/app/components/FormBuilder/UnarchiveFormButton.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	}, [id]);
+	return (
+		<Button
+			type="button"
+			variant="ghost"
+			size="lg"
+			className={cn("flex gap-2", className)}
+			onClick={handleRemove}
+		>
+			<ArchiveRestore size={12} /> Unarchive
```
Comment (kalilsn) [inline core/app/components/FormBuilder/UnarchiveFormButton.tsx:31]: Unarchive is kind of a challenging word, perhaps restore would be better?

Comment (tefkah) [reply core/app/components/FormBuilder/UnarchiveFormButton.tsx:31]: I think Restore would be better yes.

File: core/app/components/FormBuilder/actions.ts [hunk 2]
```diff
@@ -109,7 +109,16 @@ export const archiveForm = defineServerAction(async function archiveForm(id: For
...
+
+export const unarchiveForm = defineServerAction(async function unarchiveForm(id: FormsId) {
+	try {
+		await autoRevalidate(
+			db.updateTable("forms").set({ isArchived: false }).where("forms.id", "=", id)
+		).executeTakeFirstOrThrow();
+	} catch (error) {
+		return { error: "Unable to unarchive form", cause: error };
 	}
 });
```
[the remark this answers is not in the document]
Comment (tefkah) [reply core/app/components/FormBuilder/actions.ts:124]: Tracked here: https://github.com/orgs/pubpub/projects/43/views/1?query=sort%3Aupdated-desc+is%3Aopen&pane=issue&itemId=82591685

**Review by tefkah — APPROVED** [review]
Seems to work well! I agree with you that "Restore" makes more sense, and we should introduce that change here now.

It LGTM otherwise, but I would like to ask you to maybe add a test? I think you could place it here maybe, just archive and unarchive that form that is created as the test.

https://github.com/pubpub/platform/blob/447c2c2416a3a5d2a403b00f1af43c4ebe519ce1/core/playwright/formBuilder.spec.ts#L34-L47

**Review by tefkah — APPROVED** [review]
awesome, thank you for adding the test!

**Pull request merged**